src/seedu/addressbook/data/person/UniquePersonList.java

```diff
@@ -122,6 +127,37 @@ public void remove(ReadOnlyPerson toRemove) throws PersonNotFoundException {
         }
     }
 
+    /**
+     * Replaces a person's target detail with the new detail.
+     *
+     * @param toEdit person whose detail is to be edited
+     * @param detail to be replaced
+     * @param newDetail replaces the current detail
+     *
+     * @throws PersonNotFoundException if no such person could be found in the list.
+     * @throws IllegalValueException if new detail string provided is invalid.
+     */
+    public void edit(Person toEdit, String detail, String newDetail)
+            throws PersonNotFoundException, IllegalValueException {
+        if (!internalList.contains(toEdit)) {
+            throw new PersonNotFoundException();
+        }
+        switch (detail) {
+            case PERSON_DETAIL_NAME:
+                toEdit.setName(newDetail);
+                break;
+            case PERSON_DETAIL_EMAIL:
+                toEdit.setEmail(newDetail);
+                break;
+            case PERSON_DETAIL_PHONE:
+                toEdit.setPhone(newDetail);
+                break;
+            case PERSON_DETAIL_ADDRESS:
+                toEdit.setAddress(newDetail);
+                break;
+        }
```

Comment:
Maybe there should be a default case. But I suppose that is handled in the parser already if the arguments for the type of details are wrong.